File: src/main/java/com/haier/uhome/plugins/ui/Json2ClassForm.java
package com.haier.uhome.plugins.ui;

import com.haier.uhome.plugins.utils.Utils;
import com.intellij.openapi.fileChooser.FileChooserDescriptor;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.TextFieldWithBrowseButton;

import javax.swing.*;

public class Json2ClassForm {
    private JPanel rootView;
    private JCheckBox withJson;
    private JTextField className;
    private JButton generateButton;
    private JPanel groupContainer;
    private JPanel pathContainer;
    private OnGenerateClick onGenerateClick;

    public void setOnGenerateClick(OnGenerateClick onGenerateClick) {
        this.onGenerateClick = onGenerateClick;
    }

    public JPanel getRootView() {
        return rootView;
    }

    public Json2ClassForm(Project project) {
        ButtonGroup group = new ButtonGroup();
        JRadioButton dartBtn = new JRadioButton("Dart", true);
        JRadioButton javaBtn = new JRadioButton("Java");
        group.add(dartBtn);
        group.add(javaBtn);
        groupContainer.add(dartBtn);
        groupContainer.add(javaBtn);
        TextFieldWithBrowseButton textFieldWithBrowseButton = new TextFieldWithBrowseButton();
        FileChooserDescriptor fileChooserDescriptor = new FileChooserDescriptor(false, true, false,
                false, false, false);
        textFieldWithBrowseButton.addBrowseFolderListener("Save location", project.getBasePath(), null, fileChooserDescriptor);
        pathContainer.add(textFieldWithBrowseButton);
        dartBtn.setContentAreaFilled(false);
        dartBtn.setFocusPainted(false);
        javaBtn.setContentAreaFilled(false);
        javaBtn.setFocusPainted(false);
        dartBtn.addChangeListener(e -> withJson.setVisible(dartBtn.isSelected()));
        generateButton.addActionListener(e -> {
            String saveLocation = textFieldWithBrowseButton.getText().trim();
            if (Utils.isEmptyString(saveLocation)) {
                Utils.showErrorNotification(project, "save location can not be empty");
                return;
            }
            String fileName = className.getText().trim();
            if (Utils.isEmptyString(fileName)) {
                Utils.showErrorNotification(project, "class name can not be empty");
                return;
            }
            if (onGenerateClick != null) {
                onGenerateClick.onClick(saveLocation, fileName, dartBtn.isSelected(), withJson.isSelected());
            }
        });
    }


    public interface OnGenerateClick {
        void onClick(String saveLocation, String className, boolean isDart, boolean withJson);
    }


}

File: src/main/kotlin/com/zcy/plugins/ui/Json2ClassForm.kt
package com.zcy.plugins.ui

import com.intellij.openapi.fileChooser.FileChooser
import com.intellij.openapi.fileChooser.FileChooserDescriptor
import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.TextFieldWithBrowseButton
import com.intellij.openapi.vfs.VirtualFile
import com.zcy.plugins.utils.Utils
import java.awt.event.ActionEvent
import javax.swing.*

class Json2ClassForm(project: Project?) {
    lateinit var rootView: JPanel
    private lateinit var className: JTextField
    private lateinit var generateButton: JButton
    private lateinit var groupContainer: JPanel
    private lateinit var pathContainer: JPanel
    private var onGenerateClick: OnGenerateClick? = null
    private var virtualFile: VirtualFile? = null
    fun setOnGenerateClick(onGenerateClick: OnGenerateClick?) {
        this.onGenerateClick = onGenerateClick
    }

    interface OnGenerateClick {
        fun onClick(virtualFile: VirtualFile, className: String, isDart: Boolean, withJson: Boolean)
    }

    init {
        val group = ButtonGroup()
        val dartBtn = JRadioButton("Dart", true)
        val javaBtn = JRadioButton("Java")
        group.add(dartBtn)
        group.add(javaBtn)
        groupContainer.add(dartBtn)
        groupContainer.add(javaBtn)
        val textFieldWithBrowseButton = TextFieldWithBrowseButton()
        pathContainer.add(textFieldWithBrowseButton)
        dartBtn.isContentAreaFilled = false
        dartBtn.isFocusPainted = false
        javaBtn.isContentAreaFilled = false
        javaBtn.isFocusPainted = false
        textFieldWithBrowseButton.addActionListener {
            val fileChooserDescriptor = FileChooserDescriptor(
                false, true, false,
                false, false, false
            )
            virtualFile = FileChooser.chooseFile(fileChooserDescriptor, rootView, project, null)
            virtualFile?.path?.let {
                textFieldWithBrowseButton.text = virtualFile!!.path
            }
        }
        generateButton.addActionListener {
            if (null == virtualFile) {
                Utils.showErrorNotification(project, "save location can not be empty")
                return@addActionListener
            }
            val saveLocation = virtualFile?.path?.trim { it <= ' ' }
            if (Utils.isEmptyString(saveLocation)) {
                Utils.showErrorNotification(project, "save location can not be empty")
                return@addActionListener
            }
            val fileName = className.text.trim { it <= ' ' }
            if (Utils.isEmptyString(fileName)) {
                Utils.showErrorNotification(project, "class name can not be empty")
                return@addActionListener
            }
            onGenerateClick?.let {
                onGenerateClick!!.onClick(virtualFile!!, fileName, dartBtn.isSelected, true)
            }
        }
    }
}